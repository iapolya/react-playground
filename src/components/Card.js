const Card = () => {
  const { id, name, email } = this.props;
  return (
    <div className="bg-light-pink dib br3 pa3 ma2 grow pointer shadow-2">
      <img src={`https://robohash.org/${id}?set=set4&size=150x150`} alt="cat" />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
