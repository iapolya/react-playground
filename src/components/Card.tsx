import { ICat } from '../types';

interface SrcProps {
  id: number;
  size: number;
}

const getImgSrc = ({ id, size }: SrcProps) => {
  return `https://robohash.org/${id}?set=set4&size=${size}x${size}`;
};

const Card = (props: ICat) => {
  const { id, name, email } = props;

  const small = getImgSrc({ id, size: 50 });
  const medium = getImgSrc({ id, size: 100 });
  const large = getImgSrc({ id, size: 150 });

  return (
    <div className="bg-light-pink dib br3 pa3 ma2 grow pointer shadow-2">
      <img src={small} srcSet={`${small} 300w, ${medium} 768w, ${large} 1280w`} alt="cat" />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
