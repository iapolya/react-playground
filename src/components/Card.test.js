import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

it('expect to render card component', () => {
  // expect(shallow(<Card />).length).toEqual(1);
  expect(shallow(<Card id={1} email={'Lol'} name={'NAME'} />)).toMatchSnapshot();
});
