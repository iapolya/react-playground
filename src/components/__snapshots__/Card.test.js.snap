// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card component 1`] = `
<div
  className="bg-light-pink dib br3 pa3 ma2 grow pointer shadow-2"
>
  <img
    alt="cat"
    src="https://robohash.org/undefined?set=set4&size=50x50"
    srcSet="https://robohash.org/undefined?set=set4&size=50x50 300w, https://robohash.org/undefined?set=set4&size=100x100 768w, https://robohash.org/undefined?set=set4&size=150x150 1280w"
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
